contracts/account/extensions/ERC7579Modules/SocialRecoveryExecutor.sol

```diff
@@ -0,0 +1,462 @@
+        _recoveryConfigs[account].pendingExecutionHash = keccak256(executionCalldata);
+        _recoveryConfigs[account].recoveryStart = block.timestamp;
+
+        emit RecoveryStarted(account);
+    }
+
+    /// @notice Executes the recovery process after timelock period
+    /// @dev Only callable when recovery status is Ready
+    /// @param account The account to execute recovery for
+    /// @param executionCalldata The calldata to execute, must match the pending recovery digest
+    /// @custom:security Validates execution matches the hash committed during startRecovery
+    function executeRecovery(
+        address account,
+        bytes calldata executionCalldata
+    ) external virtual whenRecoveryIsReady(account) {
+        if (keccak256(executionCalldata) != _recoveryConfigs[account].pendingExecutionHash) {
+            revert ExecutionDiffersFromPending();
+        }
+
+        // reset recovery status.
+        _recoveryConfigs[account].pendingExecutionHash = bytes32(0);
+        _recoveryConfigs[account].recoveryStart = 0;
+
+        // execute the recovery.
+        // slither-disable-next-line reentrancy-no-eth
+        Address.functionCall(account, executionCalldata);
+
+        emit RecoveryExecuted(account);
+    }
+
+    /// @notice Cancels an ongoing recovery process
+    /// @dev Can only be called by the account itself
+    /// @custom:security Direct account control takes precedence over recovery process
+    function cancelRecovery() external virtual whenRecoveryIsStartedOrReady(msg.sender) {
+        _cancelRecovery(msg.sender);
+    }
+
+    /// @notice Overload: Allows guardians to cancel a recovery process
+    /// @dev Requires threshold signatures, similar to starting recovery
+    /// @param account The account to cancel recovery for
+    /// @param guardianSignatures Array of guardian signatures authorizing cancellation
+    /// @custom:security Uses same signature threshold as recovery initiation
+    function cancelRecovery(
+        address account,
+        GuardianSignature[] calldata guardianSignatures
+    ) external virtual whenRecoveryIsStartedOrReady(account) {
+        bytes32 digest = _getCancelRecoveryDigest(account, nonces(account));
+        _validateGuardianSignatures(account, guardianSignatures, digest);
+        _useNonce(account);
+
+        _cancelRecovery(account);
+    }
+
+    /// @notice Validates guardian signatures for a given digest
+    /// @dev Helper function for clients to validate signatures
+    /// @param account The ERC-7579 Account to validate signatures for
+    /// @param guardianSignatures Array of guardian signatures to validate
+    /// @param digest The digest to validate the signatures against
+    function validateGuardianSignatures(
+        address account,
+        GuardianSignature[] calldata guardianSignatures,
+        bytes32 digest
+    ) external view {
+        _validateGuardianSignatures(account, guardianSignatures, digest);
+    }
```

Comment:
What is the value of this function ?

If we want it, I'd make it "public virtual", and remove the corresponding internal one.